### PR TITLE
Move regex functions from stringutils

### PR DESCRIPTION
**Pull request by ThishaniLucas**

### CODE REVIEW

File: .github/workflows/build-master.yml [hunk 1]
```diff
@@ -0,0 +1,43 @@
+name: Build
+
+on:
+  repository_dispatch:
+    types: [ stdlib-publish-snapshot ]
+  push:
+    branches:
+      - master
```
Comment (MadhukaHarith92) [inline .github/workflows/build-master.yml:8]:
```suggestion
      - main
```

File: .github/workflows/stdlib-workflow-trigger.yml [hunk 2]
```diff
@@ -0,0 +1,24 @@
+name: Stdlib Dependency Update
+
+on:
+  push:
+    branches: 
+      - master
```
Comment (MadhukaHarith92) [inline .github/workflows/stdlib-workflow-trigger.yml:6]:
```suggestion
      - main
```

File: README.md [hunk 3]
```diff
@@ -1,2 +1,66 @@
-# module-ballerina-regex
-Ballerina Regex Module
+Ballerina Regex Library
+==============================
+
+  [![Build](https://github.com/ballerina-platform/module-ballerina-regex/workflows/Build/badge.svg)](https://github.com/ballerina-platform/module-ballerina-regex/actions?query=workflow%3ABuild)
+  [![GitHub Last Commit](https://img.shields.io/github/last-commit/ballerina-platform/module-ballerina-regex.svg)](https://github.com/ballerina-platform/module-ballerina-regex/commits/master)
```
Comment (MadhukaHarith92) [inline README.md:5]:
```suggestion
  [![GitHub Last Commit](https://img.shields.io/github/last-commit/ballerina-platform/module-ballerina-regex.svg)](https://github.com/ballerina-platform/module-ballerina-regex/commits/main)
```

File: spotbugs-exclude.xml [hunk 4]
```diff
@@ -0,0 +1,19 @@
+<!--
+  ~ Copyright (c) 2018 WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (MadhukaHarith92) [inline spotbugs-exclude.xml:2]:
```suggestion
  ~ Copyright (c) 2021 WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```

File: .github/workflows/build-master.yml (hunk 1 above)
Comment (ThishaniLucas) [reply .github/workflows/build-master.yml]: Done. Thanks for pointing out.

File: README.md [hunk 5]
```diff
@@ -1,2 +1,65 @@
...
+==============================
+
+  [![Build](https://github.com/ballerina-platform/module-ballerina-regex/workflows/Build/badge.svg)](https://github.com/ballerina-platform/module-ballerina-regex/actions?query=workflow%3ABuild)
+  [![GitHub Last Commit](https://img.shields.io/github/last-commit/ballerina-platform/module-ballerina-regex.svg)](https://github.com/ballerina-platform/module-ballerina-regex/commits/main)
+  [![Github issues](https://img.shields.io/github/issues/ballerina-platform/ballerina-standard-library/module/regex.svg?label=Open%20Issues)](https://github.com/ballerina-platform/ballerina-standard-library/labels/module%2Fregex)
+  [![License](https://img.shields.io/badge/License-Apache%202.0-blue.svg)](https://opensource.org/licenses/Apache-2.0)
+
+This Regex library is one of the standard libraries of <a target="_blank" href="https://ballerina.io/">Ballerina</a> language.
+
+For a quick sample on demonstrating the usage see [Ballerina By Example](https://ballerina.io/swan-lake/learn/by-example/).
```
Comment (daneshk) [inline README.md:11]: Shall we remove this. since we don't have a BBE to point. will add this when we add a BBE
```suggestion
```

Comment (daneshk) [reply README.md:11]: Shall we have a description of this module. Explaining what we support in this module

File: regex-ballerina/Package.md [hunk 6]
```diff
@@ -0,0 +1,3 @@
+## Module overview
+
+This module provides regex utility functions.
```
Comment (daneshk) [inline regex-ballerina/Package.md:3]: Shall we describe more about the operations we support in this module

**Review by daneshk — APPROVED** [review]

File: README.md [hunk 7]
```diff
@@ -1,2 +1,65 @@
...
-Ballerina Regex Module
+Ballerina Regex Library
+==============================
+
+  [![Build](https://github.com/ballerina-platform/module-ballerina-regex/workflows/Build/badge.svg)](https://github.com/ballerina-platform/module-ballerina-regex/actions?query=workflow%3ABuild)
+  [![GitHub Last Commit](https://img.shields.io/github/last-commit/ballerina-platform/module-ballerina-regex.svg)](https://github.com/ballerina-platform/module-ballerina-regex/commits/main)
+  [![Github issues](https://img.shields.io/github/issues/ballerina-platform/ballerina-standard-library/module/regex.svg?label=Open%20Issues)](https://github.com/ballerina-platform/ballerina-standard-library/labels/module%2Fregex)
+  [![License](https://img.shields.io/badge/License-Apache%202.0-blue.svg)](https://opensource.org/licenses/Apache-2.0)
+
+This Regex library is one of the standard libraries of <a target="_blank" href="https://ballerina.io/">Ballerina</a> language.
```
Comment (praneesha) [inline README.md:9]:
```suggestion
This Regex library is one of the standard libraries of the <a target="_blank" href="https://ballerina.io/">Ballerina</a> language.
```

File: README.md [hunk 8]
```diff
@@ -1,2 +1,65 @@
...
+
+It has support for regex functionalities like checking whether a string matches a given regex, replacing substrings and spliting strings based on a regex.
```
Comment (praneesha) [inline README.md:11]:
```suggestion
It has support for regex functionalities like checking whether a string matches a given regex, replacing substrings, and splitting strings based on a regex.
```

File: README.md [hunk 9]
```diff
@@ -1,2 +1,65 @@
...
+
+## Issues and Projects
+
+Issues and Projects tabs are disabled for this repository as this is part of the Ballerina Standard Library. To report bugs, request new features, start new discussions, view project boards, etc. please visit Ballerina Standard Library [parent repository](https://github.com/ballerina-platform/ballerina-standard-library).
```
Comment (praneesha) [inline README.md:15]:
```suggestion
The **Issues** and **Projects** tabs are disabled for this repository as this is part of the Ballerina Standard Library. To report bugs, request new features, start new discussions, view project boards, etc., go to the Ballerina Standard Library [parent repository](https://github.com/ballerina-platform/ballerina-standard-library). 
```

File: README.md [hunk 10]
```diff
@@ -1,2 +1,65 @@
...
+
+This Regex library is one of the standard libraries of <a target="_blank" href="https://ballerina.io/">Ballerina</a> language.
+
+It has support for regex functionalities like checking whether a string matches a given regex, replacing substrings and spliting strings based on a regex.
+
+## Issues and Projects
+
+Issues and Projects tabs are disabled for this repository as this is part of the Ballerina Standard Library. To report bugs, request new features, start new discussions, view project boards, etc. please visit Ballerina Standard Library [parent repository](https://github.com/ballerina-platform/ballerina-standard-library). 
+
+This repository only contains the source code for the module.
```
Comment (praneesha) [inline README.md:17]:
```suggestion
This repository contains only the source code of the module.
```

File: README.md [hunk 11]
```diff
@@ -1,2 +1,65 @@
...
+
+## Contributing to Ballerina
+
+As an open source project, Ballerina welcomes contributions from the community. 
+
+For more information, go to the [contribution guidelines](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md).
+
+## Code of Conduct
+
+All contributors are encouraged to read [Ballerina Code of Conduct](https://ballerina.io/code-of-conduct).
```
Comment (praneesha) [inline README.md:59]:
```suggestion
All contributors are encouraged to read the [Ballerina Code of Conduct](https://ballerina.io/code-of-conduct).
```

File: issue_template.md [hunk 12]
```diff
@@ -0,0 +1,18 @@
+**Description:**
+<!-- Give a brief description of the issue -->
```
Comment (praneesha) [inline issue_template.md:2]:
```suggestion
<!-- Give a brief description of the issue -->

```
Why is this file committed?

File: pull_request_template.md [hunk 13]
```diff
@@ -0,0 +1,52 @@
+## Purpose
+> Describe the problems, issues, or needs driving this feature/fix and include links to related issues in the following format: Resolves issue1, issue2, etc.
```
Comment (praneesha) [inline pull_request_template.md:2]:
```suggestion
> Describe the problems, issues, or needs driving this feature/fix and include links to related issues in the following format: Resolves issue1, issue2, etc.

```
Why is this file committed?

File: regex-ballerina/Package.md [hunk 14]
```diff
@@ -0,0 +1,8 @@
+## Module overview
+
+This module provides the following regex utility functions. 
+
+* `matches` - Checks whether a string matches a given regex.
+* `replaceAll` - Replaces every occurance of a substring that matches a given regex in a string, with another string provided.
```
Comment (praneesha) [inline regex-ballerina/Package.md:6]:
```suggestion
* `replaceAll` - Replaces every occurrence of a substring that matches a given regex in a string, with another string provided.
```

The "with another string provided" part is confusing. Can we reword this?

File: regex-ballerina/Package.md [hunk 15]
```diff
@@ -0,0 +1,8 @@
...
+* `replaceFirst` - Replaces the first occurance of a substring that matches a given regex in a string, with another string provided.
```
Comment (praneesha) [inline regex-ballerina/Package.md:7]:
```suggestion
* `replaceFirst` - Replaces the first occurrence of a substring that matches a given regex in a string, with another string provided.
```
Same as above.

File: regex-ballerina/natives.bal [hunk 16]
```diff
@@ -0,0 +1,129 @@
...
+#
+# + stringToMatch - The string to match the regex
+# + regex - The regex to match the string
+# + return - `true` if the provided string matches the regex or else
+#            `false`
+public isolated function matches(string stringToMatch, string regex) returns boolean {
+    return matchesExternal(java:fromString(stringToMatch), java:fromString(regex));
+}
+
+# Replaces each occurrence of the substrings, which matches the provided
```
Comment (praneesha) [inline regex-ballerina/natives.bal:33]:
```suggestion
# Replaces each occurrence of the substrings, which match the provided
```

File: regex-ballerina/natives.bal [hunk 17]
```diff
@@ -0,0 +1,129 @@
...
+#                 match the `regex`
+# + return - The resultant string with the replaced substrings
+public isolated function replaceAll(string originalString, string regex, string replacement) returns string {
+    handle value = replaceAllExternal(java:fromString(originalString), java:fromString(regex),
+                                      java:fromString(replacement));
+    string? updatedString = java:toString(value);
+    if (updatedString is string) {
+        return updatedString;
+    } else {
+        // should never reach here
```
Comment (praneesha) [inline regex-ballerina/natives.bal:53]:
```suggestion
        // Should never reach here.
```

File: regex-ballerina/natives.bal [hunk 18]
```diff
@@ -0,0 +1,129 @@
...
+#                 matches the `regex`
+# + return - The resultant string with the replaced substring
+public isolated function replaceFirst(string originalString, string regex, string replacement) returns string {
+    handle value = replaceFirstExternal(java:fromString(originalString), java:fromString(regex),
+                                        java:fromString(replacement));
+    string? updatedString = java:toString(value);
+    if (updatedString is string) {
+        return updatedString;
+    } else {
+        // should never reach here
```
Comment (praneesha) [inline regex-ballerina/natives.bal:79]:
```suggestion
        // Should never reach here.
```

File: pull_request_template.md (hunk 13 above)
Comment (ThishaniLucas) [reply pull_request_template.md:2]: This is a file that comes with the initial commit when creating the repo.

File: regex-ballerina/Package.md (hunk 14 above)
Comment (ThishaniLucas) [reply regex-ballerina/Package.md:6]: Just "with another string"? Basically, it'll replace the occurrences of a substring with the provided string.

**Review by praneesha — APPROVED** [review]

**Pull request merged**